### ISSUE TITLE
Dart analyzer issues

### ISSUE THREAD

**Issue by binli120** [dart-lang/sdk#36245]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA 191.6183.20`
- `2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36`
- `IU-191.6183.20, JRE 11.0.2+9-b159.22x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.14.3, screens 3840x2160`

## Exception
```
Dart analysis server, SDK version 2.2.0-edge.0a7dcf17eb5f2450480527d6ad1e201fb47f1e36, server version 1.23.0, error: Captured exception
NoSuchMethodError: The method 'refresh' was called on null.
Receiver: null
Tried calling: refresh(Instance of 'DeclarationsContext')
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
#1      DeclarationsTracker._performChangeFile (package:analyzer/src/services/available_declarations.dart:611:15)
#2      DeclarationsTracker.doWork (package:analyzer/src/services/available_declarations.dart:473:7)
#3      CompletionLibrariesWorker.performWork (package:analysis_server/src/domains/completion/available_suggestions.dart:177:13)
<asynchronous suspension>
#4      AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2125:35)
<asynchronous suspension>
#5      AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:2075:5)
#6      new AnalysisServer (package:analysis_server/src/analysis_server.dart:213:29)
#7      SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:86:26)
#8      StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:37:18)
#9      Driver.startAnalysisServer.<anonymous closure> (package:analysis_server/src/server/driver.dart:526:21)
#10     _rootRun (dart:async/zone.dart:1124:13)
#11     _CustomZone.run (dart:async/zone.dart:1021:19)
#12     _runZoned (dart:async/zone.dart:1516:10)
#13     runZoned (dart:async/zone.dart:1463:12)
#14     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:605:12)
#15     Driver.startAnalysisServer (package:analysis_server/src/server/driver.dart:524:7)
#16     Driver.start (package:analysis_server/src/server/driver.dart:427:7)
#17     main (file:///b/rr/tmpgY2JrN/w/src/third_party/dart/pkg/analysis_server/bin/server.dart:12:11)
#18     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#19     main (file:///b/rr/tmpgY2JrN/w/src/third_party/dart/pkg/analysis_server/bin/server.dart:10:10)
#20     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:298:32)
#21     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

For additional log information, please append the contents of
file:///private/var/folders/js/vp03v3hd49xgz1dh99l8dg200000gp/T/report.txt.

**Comment by DanTup** [comment]
Are you using a Dart nightly build? I think upgrading to a more recent one will fix it - it looks like #36212.

**Comment by scheglov** [comment]
https://github.com/dart-lang/sdk/issues/36212
